backend/backend.go

```diff
@@ -540,6 +544,97 @@ func (backend *Backend) Accounts() AccountsList {
 	return backend.accounts
 }
 
+// AccountsByKeystore returns a map of the current accounts of the backend, grouped
+// by keystore.
+func (backend *Backend) AccountsByKeystore() (KeystoresAccountsListMap, error) {
+	accountsByKeystore := KeystoresAccountsListMap{}
+	defer backend.accountsAndKeystoreLock.RLock()()
+	for _, account := range backend.accounts {
+		persistedAccount := account.Config().Config
+		rootFingerprint, err := persistedAccount.SigningConfigurations.RootFingerprint()
+		if err != nil {
+			return nil, err
+		}
+		keystore, err := backend.Config().AccountsConfig().LookupKeystore(rootFingerprint)
+		if err != nil {
+			return nil, err
+		}
+		accountsByKeystore[keystore] = append(accountsByKeystore[keystore], account)
+	}
+	return accountsByKeystore, nil
+}
+
+// KeystoreTotalAmount represents the total balance amount of the accounts belongings to a keystore.
+type KeystoreTotalAmount = struct {
+	// FiatUnit is the fiat unit of the total amount
+	FiatUnit string `json:"fiatUnit"`
+	// Total formatted for frontend visualization
+	Total string `json:"total"`
+}
+
+// AccountsTotalBalanceByKeystore returns a map of accounts' total balances across coins, grouped by keystore.
+func (backend *Backend) AccountsTotalBalanceByKeystore() (map[string]KeystoreTotalAmount, error) {
+	totalAmounts := make(map[string]KeystoreTotalAmount)
+	fiat := backend.Config().AppConfig().Backend.MainFiat
+	isFiatBtc := fiat == rates.BTC.String()
+	fiatUnit := fiat
+	if isFiatBtc && backend.Config().AppConfig().Backend.BtcUnit == coin.BtcUnitSats {
+		fiatUnit = "sat"
+	}
+
+	formatBtcAsSat := util.FormatBtcAsSat(backend.Config().AppConfig().Backend.BtcUnit)
+
+	accountsByKeystore, err := backend.AccountsByKeystore()
+	if err != nil {
+		return nil, err
+	}
+	for keystore, accountList := range accountsByKeystore {
```

Comment:
a lot of duplication here with chart.go. The total computed in chart.go is basically the sum of the keystore totals here right? seems like code re-use could be possible here, or am i missing something?

no need to do it in this PR, but it would be nice to de-duplicate.

Reply:
Addressed in #2609